File: frontend/src/types/interview.ts
import { Question } from "./question";

export type SessionMode = "flashcard" | "mock";
export type SessionStatus = "active" | "completed";

export interface STARBreakdown {
  situation: number;
  task: number;
  action: number;
  result: number;
}

export interface Feedback {
  overall_score: number;
  star_breakdown: STARBreakdown;
  strengths: string[];
  improvements: string[];
  suggested_revision?: string;
  follow_up_question?: string;
}

export interface AnswerRecord {
  id: string;
  question_id: string;
  question_text: string;
  answer_text: string;
  overall_score?: number;
  feedback?: Feedback;
  created_at: string;
}

export interface InterviewSession {
  id: string;
  mode: SessionMode;
  background_id: string;
  job_description_id?: string;
  questions: Question[];
  current_question_index: number;
  status: SessionStatus;
  created_at: string;
}

export interface StartSessionRequest {
  mode: SessionMode;
  background_id: string;
  job_description_id?: string;
  question_count: number;
}

export interface SessionSummary {
  session_id: string;
  total_questions: number;
  questions_answered: number;
  questions_skipped: number; // NEW FIELD
  average_score?: number;
  answers: AnswerRecord[];
}

// NEW TYPES
export interface SkipQuestionResponse {
  skipped: boolean;
  question_index: number;
  has_more_questions: boolean;
  next_question?: Question;
}

export interface GenerateMoreResponse {
  new_questions: Question[];
  total_questions: number;
  current_question_index: number;
}

export interface NextQuestionResponse {
  completed: boolean;
  no_more_questions?: boolean;
  question_index?: number;
  question?: Question;
  message?: string;
  total_questions?: number;
}

File: backend/app/schemas/interview.py
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum
from datetime import datetime


class SessionMode(str, Enum):
    FLASHCARD = "flashcard"
    MOCK = "mock"


class SessionStatus(str, Enum):
    ACTIVE = "active"
    COMPLETED = "completed"


class StartSessionRequest(BaseModel):
    mode: SessionMode
    background_id: str
    job_description_id: Optional[str] = None
    question_count: int = 5


class AnswerSubmission(BaseModel):
    answer_text: str
    audio_duration_seconds: Optional[float] = None


class STARBreakdown(BaseModel):
    situation: int
    task: int
    action: int
    result: int


class FeedbackResponse(BaseModel):
    overall_score: int
    star_breakdown: STARBreakdown
    strengths: List[str]
    improvements: List[str]
    suggested_revision: Optional[str] = None
    follow_up_question: Optional[str] = None


class AnswerRecordResponse(BaseModel):
    id: str
    question_id: str
    question_text: str
    answer_text: str
    overall_score: Optional[int]
    feedback: Optional[FeedbackResponse]
    created_at: datetime


class SessionResponse(BaseModel):
    id: str
    mode: SessionMode
    background_id: str
    job_description_id: Optional[str]
    questions: List[dict]
    current_question_index: int
    status: SessionStatus
    created_at: datetime


class SessionSummary(BaseModel):
    session_id: str
    total_questions: int
    questions_answered: int
    average_score: Optional[float]
    answers: List[AnswerRecordResponse]